src/test/java/sqlline/SqlLineArgsTest.java

```diff
@@ -2225,6 +2225,102 @@ public void testStartupArgsWithoutUrl() {
     }
   }
 
+  /**
+   * Test to Set/Reset confirm.
+   */
+  @Test
+  public void testConfirmFlag() {
+    try {
+      DispatchCallback callback = new DispatchCallback();
+      SqlLine line = new SqlLine();
+      ByteArrayOutputStream os = new ByteArrayOutputStream();
+      assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+      begin(line, os, false, "-e", "!set confirm true");
+      assertEquals(Boolean.TRUE, line.getOpts().getConfirm());
+      begin(line, os, false, "-e", "!set confirm false");
+      assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+    } catch (Throwable e) {
+      //fail
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Test effects of confirm flag.
+   */
+  @Test
+  public void testConfirmFlagEffects() {
+    new MockUp<sqlline.Commands>() {
+      @Mock
+      private int getUserAnswer(
+              String question, int... allowedAnswers) throws IOException {
+        return 'y';
+      }
+    };
+    SqlLine line = new SqlLine();
+
+    assertEquals(Boolean.FALSE, line.getOpts().getConfirm());
+    assertEquals("default", line.getOpts().getConfirmPattern());
+    String script = "CREATE TABLE dummy_test(pk int);\n"
+            + "INSERT INTO dummy_test(pk) VALUES(1);\n"
+            + "INSERT INTO dummy_test(pk) VALUES(2);\n"
+            + "INSERT INTO dummy_test(pk) VALUES(3);\n"
+            + "DELETE FROM dummy_test;\n"
+            + "DROP TABLE dummy_test;\n";
+    checkScriptFile(script, true,
+            equalTo(SqlLine.Status.OK),
+            containsString(" "));
+    script = "!set confirm true\n" + script;
+    checkScriptFile(script, true,
+            equalTo(SqlLine.Status.OK),
+            containsString(" "));
+
```

Comment:
Shouldn't we have to verify interaction/non-interaction at least with the mocks for handling the delete/drops?